sklearn/metrics/classification.py

```diff
@@ -1225,14 +1250,45 @@ class 2       1.00      0.67      0.80         3
     else:
         labels = np.asarray(labels)
 
-    last_line_heading = 'avg / total'
+    is_binary_classification = len(labels) == 2
+    is_multilabel = _check_targets(y_true, y_pred)[0].startswith('multilabel')
+
+    if is_binary_classification and average:
+        raise ValueError('Averaging methods are not relevant for binary '
+                         'classification. Please omit average argument.')
```

Comment:
I wouldn't raise a value error here, as you can consider binary classification as multiclass task with two class. You will get the relevant information for binary task if you have a look only at the metric of each label.